File: test/Test_PlatypusEgg.cpp
#include "../include/PlatypusEgg.cpp"
#include <gtest/gtest.h>
#include <string>

using namespace std;

TEST(PlatypusEggTest, AllFunctionalityofPlatypusEgg){
    PlatypusEgg testPE(40, "Masku");
    ASSERT_EQ("Masku", testPE.getProducer());
    ASSERT_EQ(PLATYPUS_EGG, testPE.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testPE.getProductType());
    ASSERT_EQ(40, testPE.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/egg/TestPlatypusEgg.java
package engisubs.gameobject.product.farmproduct.egg;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestPlatypusEgg{
    private PlatypusEgg testPlatypusEgg;
    @Before
    public void initObject(){
        testPlatypusEgg = new PlatypusEgg(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", testPlatypusEgg.getProducer());
        Assert.assertEquals(GameObjectType.PLATYPUS_EGG, testPlatypusEgg.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, testPlatypusEgg.getProductType());
        Assert.assertEquals(25, testPlatypusEgg.getPrice());
    }

}